### ISSUE TITLE
How to set up remote MySQL DB/settings for Compendium?

### ISSUE THREAD

**Issue by alexbfree** [CompendiumNG/CompendiumNG#271]
Hi
I have created a MySQL on on my webhost (HostGator) and enabled my IP for remote access.
When I try to connect in CompendiumNG v2.1.4 I get an error when I try to connect. 

```
Error Connecting to MySQL database MySQLDB
Access denied for user 'abowyer_compend'@'128.240.225.72' to database 'mysql'.
```

How is this supposed to work? It seems to be trying to connect to a database named `mysql` but that is not the name of my DB. In fact that is not a legal name on my web host, it has to be `abowyer_databasename`. But there doesn't seem to be any way in CompendiumNG to enter the database name.

Am I completely blocked from using this feature, or am I missing something?

Can you add a database name field to the settings pane? Or can I specify it somewhere else in a settings file or something?

**Comment by maiklos** [comment]
isn't there something above/below the message ?

**Comment by alexbfree** [comment]
no... 

![screenshot 2016-11-23 15 43 29](https://cloud.githubusercontent.com/assets/1473244/20568216/a2977906-b193-11e6-909b-0cf010d5add4.png)

I have no database called 'mysql' though so this could be the issue couldn't it?

**Comment by alexbfree** [comment]
@maiklos any word on this? If you can't suggest a fix for my issue, can you provide a working example of using Compendium with a remote MySQL server with a specific database name? Or does it only support remote MySQL where it can create its own DBs? I notice that in my local MySQL based Compendium install it has created two databases - `cngadmindb` and `<MyProjectName>_1479918423987` .. Is it perhaps trying to do this on the remote server which it doesn't have permission to do - I am sure you can appreciate this will be the case with any commercial DB host...

Is there a config file or code file somewhere where I can override the hardcoding of database names?

**Comment by maiklos** [comment]
@alexbfree bit late to the party but anyway: is this still an issue for you ?

**Comment by alexbfree** [comment]
Yes I never did find a solution, would still like a way to do this.
Cheers

**Comment by maiklos** [comment]
Could you please verify in cPanel that you external IP is whitelisted  as described here: [http://support.hostgator.com/articles/cpanel/how-to-whitelist-your-ip-in-cpanel-for-remote-mysql-access](http://support.hostgator.com/articles/cpanel/how-to-whitelist-your-ip-in-cpanel-for-remote-mysql-access) ?

**Comment by alexbfree** [comment]
There's no point - that database that Compendium is trying to access does not exist, if the error message is to be believed.

The issue is surely that my database name is not `mysql`, a name which Hostgator (and many other hosts) would not allow me to use (since all DB names must be prefixed by the account name) and the Compendium software doesn't seem to allow me to specify the DB name.

**Comment by alexbfree** [comment]
(please re-read my initial comment thread above for more detail on the DB naming issue)

**Comment by maiklos** [comment]
ok, that's probably true. Database mysql IMO must exist.. it is just that your user doesn't have access to it. I was under impression that you have dedicated Mysql server - which is apparently not the case.

I am not sure where is the database name (mysql) taken from. I suspect that it might be stored in your derby admin db. Could you connect to your local cngadmindb db (user should be root and password empty) and look what you have in "connections" table in fied DEFAULTDATABASE ? 

hint for connecting to it:
path should be your home dir .... compendiumng_data/cngadmindb
user: root
password: -empty-
...and compendiumng app must be switched off.

also please check if you have MySQL.properties file in your home directory under compendiumng_config and what is in it.

**Comment by alexbfree** [comment]
I have found the directory `/Applications/CompendiumNG/data/cngadmindb` but can't seem to find the right MySQL incantation to let me connect. Getting connection refused - not sure if I am doing something wrong with incantation, or with path specification. If you know the right command to connect that would be great.

I don't seem to have a MySQL.properties in my home directory, nor a compendiumng_config directory - is that a Windows only thing perhaps? I am on Mac. Or did you mean I should be checking on the remote server?

**Comment by mike223** [comment]
Not sure if this is related:
Under CNG 2.1.5   
  ~/compendiumng_config/Format.properties
contains a setting 
  database=mysql
When I changed "mysql" to something else CNG started with the derby database instead of MySQL. 

From the previous Compendium 2.0 Beta 1 I remember that you not only needed the database but almost all permissions on it too.
